lightning/src/onion_message/messenger.rs

```diff
@@ -432,6 +432,22 @@ pub enum MessageSendInstructions {
 	},
 }
 
+/// A placeholder value used for paths within the codebase.
+///
+/// PATHS_PLACEHOLDER` is temporarily used as a default value in situations
+/// where a path index is required but has not yet been assigned or initialized.
+pub const PATHS_PLACEHOLDER: usize = 0;
+
+/// Represents the types of [`BlindedMessagePath`] that can be created.
+///
+/// This struct allows to specify the properties of a Blinded Path before creating it.
+pub struct BlindedPathParams {
+	/// Number of [`BlindedMessagePath`]s to be created.
+	pub paths: usize,
```

Comment:
Paths aren't only for QR codes. We use them in reply paths and in messages not transmitted via QR code. The router doesn't know this context.

Reply:
I guess the `MessageRouter` does get the `MessageContext` now, though that puts the onus on the `MessageRouter` implementation to take the correct action. Maybe that's ultimately desirable? Not sure given you would need to reimplement the logic in `DefaultMessageRouter`.

(Overriding `MessageRouter` is a little annoying in that `ChannelManager` has a `Router` which also requires `MessageRouter`, and `DefaultRouter` also wraps `DefaultMessageRouter`. So it currently requires making a `Router` that wraps `DefaultRouter` and your own `MessageRouter`, implementing the latter by delegating to it as needed. Making the traits independent would make this less cumbersome, if adding another type parameter to `ChannelManager`.)

I think we still want a way for the user to say they don't want any blinded paths in an offer or refund, though.

Reply:
I'm not sure I'm convinced asking the user re-implement the `MessageRouter` trait is really that trivially. Not only given the above, but for example if they are sure they want to create an offer with one blinded path (perhaps for a QR code) and another offer with many blinded paths (perhaps for BIP 353). Maybe they call a method on their `MessageRouter` that puts it in a different mode before calling `pay_for_offer`? It doesn't seem as intuitive as passing a parameter to `pay_for_offer`.

Reply:
With the [pr3246.07](https://github.com/shaavan/rust-lightning/commits/pr3246.07) I have updated the struct to be simpler. So that maintains the control to MessageRouter to decide how the BlindedPath should look!
So I am marking this as resolved for now!
Thank you so much!